meson.build

```diff
@@ -0,0 +1,209 @@
+project('libuv', 'c', version : '1.18.0', license : 'libuv')
+
+cc = meson.get_compiler('c')
+
+uvsrc = [
+  'src/fs-poll.c',
+  'src/inet.c',
+  'src/threadpool.c',
+  'src/uv-common.c',
+  'src/version.c',
+  'src/unix/async.c',
+  'src/unix/core.c',
+  'src/unix/dl.c',
+  'src/unix/fs.c',
+  'src/unix/getaddrinfo.c',
+  'src/unix/getnameinfo.c',
+  'src/unix/loop-watcher.c',
+  'src/unix/loop.c',
+  'src/unix/pipe.c',
+  'src/unix/poll.c',
+  'src/unix/process.c',
+  'src/unix/signal.c',
+  'src/unix/stream.c',
+  'src/unix/tcp.c',
+  'src/unix/thread.c',
+  'src/unix/timer.c',
+  'src/unix/tty.c',
+  'src/unix/udp.c',
+  'src/unix/linux-core.c',
+  'src/unix/linux-inotify.c',
+  'src/unix/linux-syscalls.c',
+  'src/unix/procfs-exepath.c',
+  'src/unix/proctitle.c',
+  'src/unix/sysinfo-loadavg.c',
+  'src/unix/sysinfo-memory.c',
+]
+
+uvdefines = [ '-D_GNU_SOURCE', ]
+uvincdir = include_directories('include', 'src', 'src/unix')
+
+if cc.get_define('__ANDROID__') != ''
+  uvsrc += [ 'src/unix/android-ifaddrs.c',
+             'src/unix/pthread-fixes.c' ]
+endif
+
+pthread = cc.find_library('pthread', required: true)
+
+libuv_deps = [ cc.find_library('m', required: true),
+               cc.find_library('dl', required: true),
```

Comment:
These should both be `required : false`. Most platforms have these inside libc and will fail if these are set to true.